src/msw/uilocale.cpp

```diff
@@ -55,6 +56,40 @@ static void wxMSWSetThreadUILanguage(LANGID langid)
     }
 }
 
+// Trivial wrapper for ::CompareStringEx().
+//
+// TODO-XP: Drop this when we don't support XP any longer.
+static int wxMSWCompareStringEx(LPCWSTR lpLocaleName,
+                                 DWORD dwCmpFlags,
+                                 LPCWSTR lpString1, //_In_NLS_string_(cchCount1)LPCWCH lpString1,
+                                 int cchCount1,
+                                 LPCWSTR lpString2, //_In_NLS_string_(cchCount2)LPCWCH lpString2,
+                                 int cchCount2,
+                                 LPNLSVERSIONINFO lpVersionInformation,
+                                 LPVOID lpReserved,
+                                 LPARAM lParam)
+{
+    // Avoid calling CompareStringEx() on XP.
+    if ( wxGetWinVersion() >= wxWinVersion_Vista )
+    {
+        
+        typedef int(WINAPI *CompareStringEx_t)(LPCWSTR,DWORD,LPCWSTR,int,LPCWSTR,int,LPNLSVERSIONINFO,LPVOID,LPARAM);
+        static const CompareStringEx_t INVALID_FUNC_PTR = (CompareStringEx_t)-1;
+
+        static CompareStringEx_t pfnCompareStringEx = INVALID_FUNC_PTR;
+        if (pfnCompareStringEx == INVALID_FUNC_PTR)
+        {
+            wxLoadedDLL dllKernel32(wxS("kernel32.dll"));
+            wxDL_INIT_FUNC(pfn, CompareStringEx, dllKernel32);
+        }
+
+        if (pfnCompareStringEx)
+            return pfnCompareStringEx(lpLocaleName, dwCmpFlags, lpString1, cchCount1, lpString2,
+                                      cchCount2, lpVersionInformation, lpReserved, lParam);
```

Comment:
Very minor style comment:
```suggestion
        {
            return pfnCompareStringEx(lpLocaleName, dwCmpFlags, lpString1, cchCount1, lpString2,
                                      cchCount2, lpVersionInformation, lpReserved, lParam);
        }
```
(braces should be used around multiline compound statement bodies for readability).